server/lib/chat/chat-service.ts

```diff
@@ -257,6 +251,11 @@ export default class ChatService {
 
             try {
               userChannelEntry = await addUserToChannel(userId, channel.id, client)
+              if (!userChannelEntry) {
```

Comment:
This feels fairly unwieldy, adding more booleans for mutually exclusive conditions doesn't feel great. Imo you should have an enum/union type of some sort and combine all 3 exit conditions into one variable.

As evidence for this being a problem, you added these condition variables but did not add them as a loop exit condition, so this loop will always run the maximum number of times in the case that they're at a limit.